modules/dxp/apps/portal-search-tuning/portal-search-tuning-blueprints-admin-web/src/main/resources/META-INF/resources/js/edit_blueprint/PreviewSidebar.js

```diff
@@ -53,6 +53,30 @@ function PreviewSidebar({
 		_handleFetch();
 	}, [activeDelta, activePage]);
 
+	const _copyResultsToClipboard = () => {
+		navigator.clipboard.writeText(JSON.stringify(results, null, 2));
+
+		openSuccessToast({
+			message: Liferay.Language.get('copied-to-clipboard'),
+		});
+	};
+
+	const _downloadResultsToTxtFile = () => {
+		const element = document.createElement('a');
+		element.href = URL.createObjectURL(
+			new Blob([JSON.stringify(results, null, 2)], {
+				type: 'text/plain',
+			})
+		);
+		element.download = 'results.txt';
+		document.body.appendChild(element);
+		element.click();
```

Comment:
Remember to add new line breaks where sorting shouldn't happen.

Also this seems strange that we're adding an element to the body and then clicking it? It doesn't seem like good practice to generating empty elements on the body element for the only purpose of clicking only once. Because if you were click the download button multiple times, we'd be generating a lot of empty elements?

If we want a clickable link we should render a `ClayLink`. There's a prop to style links as buttons. https://clayui.com/docs/components/link/api.html